shared-libs/cht-datasource/src/person.ts

```diff
@@ -109,5 +92,50 @@ export namespace v1 {
    * @returns a function for retrieving a paged array of people
    * @throws Error if a data context is not provided
    */
-  export const getPage = getPeople(Local.Person.v1.getPage, Remote.Person.v1.getPage);
+  export const getPage = (
+    context: DataContext
+  ): ({ personType, limit, skip }: GetPageParams) => Promise<Person[]> => {
+    assertDataContext(context);
+    const fn = adapt(context, Local.Person.v1.getPage, Remote.Person.v1.getPage);
+
+    /**
+     * Returns an array of people for the provided page specifications.
+     * @param params the function params
+     * @param params.personType the type of people to return
+     * @param params.limit the maximum number of people to return. Default is 100.
+     * @param params.skip the number of people to skip. Default is 0.
+     * @returns an array of people for the provided page specifications.
+     * @throws Error if no type is provided or if the type is not for a person
+     * @throws Error if the provided `limit` value is `<=0`
+     * @throws Error if the provided `skip` value is `<0`
+     */
+    const curriedFn = async ({ personType, limit = 100, skip = 0}: GetPageParams): Promise<Person[]> => {
+      assertTypeQualifier(personType);
+      assertLimit(limit);
+      assertSkip(skip);
+
+      return fn(personType, limit, skip);
+    };
+    return curriedFn;
+  };
+
+  /**
+   * Returns a generator function for retrieving people from the given data context.
+   * @param context the current data context
+   * @returns a generator function for retrieving people
+   * @throws Error if a data context is not provided
+   */
+  export const getAll = (
+    context: DataContext
+  ): (personType: ContactTypeQualifier) => AsyncGenerator<v1.Person, void> => {
+    assertDataContext(context);
+
+    return async function* (personType: ContactTypeQualifier): AsyncGenerator<Person, void> {
```

Comment:
As with the `curriedFn` in  `getPage`, I think we should try to get some docs for this function to show up!  

```ts
    /**
     * Returns a generator for fetching all people with the given type
     * @param personType the type of people to return
     * @returns a generator for fetching all people with the given type
     * @throws Error if no type is provided or if the type is not for a person
     */
```

Reply:
I tried adding the docs to this curried function with and without the pattern we are doing for the other ones but it didn't work. I think it's because it's a generator function and I didn't want to spend too much time on this.

Reply:
I managed to get this to work!  The key is, as I mentioned above, you have to set the return type of the outer function to be `typeof` the curried function variable name.  (for some reason this works thanks to the fact that types are not real... :shrug: )

```ts
  /**
   * Returns a generator function for retrieving people from the given data context.
   * @param context the current data context
   * @returns a generator function for retrieving people
   * @throws Error if a data context is not provided
   */
  export const getAll = (
    context: DataContext
  ): typeof curriedGen => {
    assertDataContext(context);

    /**
     * Returns a generator for fetching all people with the given type
     * @param personType the type of people to return
     * @returns a generator for fetching all people with the given type
     * @throws Error if no type is provided or if the type is not for a person
     */
    const curriedGen = async function* (personType: ContactTypeQualifier): AsyncGenerator<Person, void> {
      assertTypeQualifier(personType);
      const getPage = context.bind(v1.getPage);
      const limit = 100;
      const skip = 0;
      yield* getDocumentStream(getPage, { personType, limit, skip });
    };
    return curriedGen;
  };
```

Reply:
ahhh! excellent find. Thank you.